### ISSUE TITLE
what's the target? Action Recognition or others?

### ISSUE THREAD

**Issue by ChengYiBin** [henrypowell87/2-Layer-LSTM-for-Human-Movement-Analysis#1]
Hello, I see this line of code in line 60:
`model.add(keras.layers.Dense(n_outputs, activation=tf.nn.sigmoid))`
It means u just do a binary classification, not action recognition (multi-class classification) ?
Is my understanding wrong?
looking forward to your reply :)

**Comment by henrypowell87** [comment]
Action recognition can be either a multiclass or a binary class classification problem. It just depends on how many different actions you want to have the classifier differentiate. This code uses a binary cross entropy loss which means it's designed to work on binary classification problems. It was written to work in conjuction with skeleton (3D position) data from the NTURGB+D dataset which you can find here (https://github.com/shahroudy/NTURGB-D). When I wrote the code I was interested in testing the performance of the model on a simple problem so I reduced the data from a 60 class data set to a two class data set hence the use of the binary cross entropy loss.

If you want to change the model to work on a multi class problem I would recommend using a categorical cross entropy loss if your labels are one hot encoded:

```
model.compile(optimizer='adam',
                  loss='categorical_crossentropy',
                  metrics=['acc']
```

Or a sparse categorical crossentropy if your labels are integer encoded:


```
model.compile(optimizer='adam',
                  loss='sparse_categorical_crossentropy',
                  metrics=['sparse_categorical_accuracy']
```

Note in the last instance that the accuracy metric has changed.
I hope this answers your question.

**Comment by ChengYiBin** [comment]
Thanks! your answer helps a lot!! And it's the same as what I thought.
By the way, as you mentioned that u reduced the number of class from 60 to 2, I'm curious if you reduced them to single action (means single person behaves) or interaction (means action performed by multiple person)
Thanks your answer again!!! :)

**Comment by henrypowell87** [comment]
They were both single person actions (brushing teeth, and drinking from a cup).
